setup.py

```diff
@@ -2,6 +2,22 @@
 
 from setuptools import setup, find_packages
 
+extra = {}
+
+try:
+    import Cython.Build
+    # cython didn't support "from __future__" statements until 0.17.
+    if tuple(int(v) for v in Cython.__version__.split('.')) > (0, 17, 0):
+        extra = {
+            'ext_modules': Cython.Build.cythonize([
+                'thriftrw/protocol/binary.py',
+                'thriftrw/wire/value.py',
+                'thriftrw/spec/*.py',
+            ]),
+        }
+except ImportError:
+    pass
```

Comment:
I think what you want to do instead is rename the files you want to compile to `.pyx` and always compile them to C if cython is present. If it's not present, we should assume that the C files are present (they will need to be added to MANIFEST.in), and add them to `ext_modules` regardless. See http://stackoverflow.com/a/4515279/58536.

That way users don't have to depend on cython. We'll only need it during development and release.